website/addons/dataverse/static/dataverseFangornConfig.js

```diff
@@ -33,45 +22,53 @@ function _fangornActionColumn (item, col) {
+            m('h3', 'Publish this ' + toPublish + '?'),
+            m('p.m-md', both ? 'This dataset cannot be published until ' + item.data.dataverse + ' Dataverse is published. ' : ''),
+            m('p.m-md', 'By publishing this ' + toPublish + ', all content will be made available through the Harvard Dataverse using their internal privacy settings, regardless of your OSF project settings. '),
+            m('p.font-thick.m-md', both ? 'Do you want to publish this Dataverse AND this dataset?' : 'Are you sure you want to publish this dataset?')
         ];
         var modalActions = [
             m('button.btn.btn-default.m-sm', { 'onclick' : function (){ self.modal.dismiss(); }},'Cancel'),
-            m('button.btn.btn-primary.m-sm', { 'onclick' : function() { releaseStudy(); } }, 'Release Study')
+            m('button.btn.btn-primary.m-sm', { 'onclick' : function() { publishDataset(); } }, 'Publish ' + toPublish)
         ];
 
         this.modal.update(modalContent, modalActions);
 
-        function releaseStudy() {
+        function publishDataset() {
             self.modal.dismiss();
-            item.notify.update('Releasing Study', 'info', 1, 3000);
+            item.notify.update('Publishing ' + toPublish, 'info', 1, 3000);
             $.osf.putJSON(
                 url,
                 {}
             ).done(function(data) {
                 var modalContent = [
-                    m('p.m-md', 'Your study has been released. Please allow up to 24 hours for the released version to appear on your OSF project\'s file page.')
+                    m('p.m-md', 'Your content has been published.')
                 ];
                 var modalActions = [
                     m('button.btn.btn-primary.m-sm', { 'onclick' : function() { self.modal.dismiss(); } }, 'Okay')
                 ];
                 self.modal.update(modalContent, modalActions);
-            }).fail( function(args) {
-                var message = args.responseJSON.code === 400 ?
-                    'Error: Something went wrong when attempting to release your study.' :
-                    'Error: This version has already been released.';
+                item.data.state = 'published';
+                item.data.hasPublishedFiles = item.children.length > 0;
+            }).fail(function(args) {
+                var statusCode = args.responseJSON.code;
+                var message;
+                switch (statusCode) {
+                    case 405:
+                        message = 'Error: This dataset cannot be published until ' + item.data.dataverse + ' Dataverse is published.';
+                        break;
+                    case 409:
+                        message = 'This dataset version has already been published.';
+                        break;
+                    default:
+                        message = 'Error: Something went wrong when attempting to publish your dataset.';
```

Comment:
In general, we log anything that is unexpected. We could also get here if an uncaught exception occurred in our server code.